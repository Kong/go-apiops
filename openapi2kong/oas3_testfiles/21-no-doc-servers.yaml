# openapi2kong.go should not create an extra service
# when there are no top-level doc servers defined.

openapi: 3.0.0
info:
  title: Multi-Service API
  version: 1.0.0

paths:
  /users:
    get:
      servers:
        - url: https://user-service.example.com
      responses:
        '200':

  /orders:
    post:
      servers:
        - url: https://order-service.example.com
      responses:
        '201':