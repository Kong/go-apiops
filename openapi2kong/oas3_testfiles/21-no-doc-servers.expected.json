{
  "_format_version": "3.0",
  "services": [
    {
      "host": "order-service.example.com",
      "id": "8b6ad746-12b9-50fd-8604-5e7768c4f5e9",
      "name": "multi-service-api_orders_post",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "e8d42325-3ee0-57b3-9883-5ea51618f4ac",
          "methods": [
            "POST"
          ],
          "name": "multi-service-api_orders_post",
          "paths": [
            "~/orders$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_21-no-doc-servers.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_21-no-doc-servers.yaml"
      ]
    },
    {
      "host": "user-service.example.com",
      "id": "6e5e1d3b-894b-5ce8-b807-532df7b6d01d",
      "name": "multi-service-api_users_get",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "744254cb-7b0f-527d-a9e3-41998313e8fd",
          "methods": [
            "GET"
          ],
          "name": "multi-service-api_users_get",
          "paths": [
            "~/users$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_21-no-doc-servers.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_21-no-doc-servers.yaml"
      ]
    }
  ],
  "upstreams": []
}